tests/runtime/processor_labels.c

```diff
@@ -0,0 +1,563 @@
+
+    out_ffd = flb_output(ctx, (char *) "lib", &cb_data);
+    TEST_CHECK(out_ffd >= 0);
+    flb_output_set(ctx, out_ffd, "match", "test", NULL);
+
+
+    ret = flb_start(ctx);
+    TEST_CHECK(ret == 0);
+
+    flb_time_msleep(1500); /* waiting flush */
+
+    flb_stop(ctx);
+    flb_destroy(ctx);
+}
+
+
+static int cb_hash_labels(void *record, size_t size, void *data)
+{
+    int ret;
+    size_t off = 0;
+    cfl_sds_t text = NULL;
+    struct cmt *cmt = NULL;
+
+    /* get cmetrics context */
+    ret = cmt_decode_msgpack_create(&cmt, (char *) record, size, &off);
+    if (ret != 0) {
+        flb_error("could not process metrics payload");
+        return -1;
+    }
+
+    /* convert to text representation */
+    text = cmt_encode_text_create(cmt);
+    TEST_CHECK(text != NULL);
+
+    /* it should only update the metrics which contains a label with the name 'hostname' */
+    TEST_CHECK(count_metrics_matches(text, "hostname=\"49960de5880e8c687434170f6476605b8fe4aeb9a28632c7995cf3ba831d9763\"") == 2);
+
+    if (record) {
+        flb_free(record);
+    }
+
+    /* destroy cmt context */
+    cmt_destroy(cmt);
+    cmt_encode_text_destroy(text);
+
+    return 0;
+}
+
+static void hash_label()
+{
+    int ret;
+    flb_ctx_t *ctx;
+    int in_ffd;
+    int out_ffd;
+    struct flb_processor *proc;
+    struct flb_processor_unit *pu;
+    struct flb_lib_out_cb cb_data;
+
+    struct cfl_variant var_static = {
+        .type = CFL_VARIANT_STRING,
+        .data.as_string = "hostname",
+    };
+
+    /* Prepare output callback with expected result */
+    cb_data.cb = cb_hash_labels;
+    cb_data.data = NULL;
+
+    /* Create context */
+    ctx = flb_create();
+    flb_service_set(ctx,
+                    "Flush", "0.200000000",
+                    "Grace", "2",
+                    NULL);
+
+    /* Input */
+    in_ffd = flb_input(ctx, (char *) "event_type", NULL);
+    TEST_CHECK(in_ffd >= 0);
+    ret = flb_input_set(ctx, in_ffd, "tag", "test", NULL);
+    TEST_CHECK(ret == 0);
+    ret = flb_input_set(ctx, in_ffd, "type", "metrics", NULL);
+    TEST_CHECK(ret == 0);
+    ret = flb_input_set(ctx, in_ffd, "interval_sec", "1", NULL);
+    TEST_CHECK(ret == 0);
+
+    /* Processor */
+    proc = flb_processor_create(ctx->config, "unit_test", NULL, 0);
+    TEST_CHECK(proc != NULL);
+
+    pu = flb_processor_unit_create(proc, FLB_PROCESSOR_METRICS, "labels");
+    TEST_CHECK(pu != NULL);
+
+    ret = flb_processor_unit_set_property(pu, "hash", &var_static);
+    TEST_CHECK(ret == 0);
+
+    /* set up processor */
+    ret = flb_input_set_processor(ctx, in_ffd, proc);
+    TEST_CHECK(ret == 0);
+
+    out_ffd = flb_output(ctx, (char *) "lib", &cb_data);
+    TEST_CHECK(out_ffd >= 0);
+    flb_output_set(ctx, out_ffd, "match", "test", NULL);
+
+
+    ret = flb_start(ctx);
+    TEST_CHECK(ret == 0);
+
+    flb_time_msleep(1500); /* waiting flush */
+
+    flb_stop(ctx);
+    flb_destroy(ctx);
+}
+
+TEST_LIST = {
+#ifdef FLB_HAVE_METRICS
```

Comment:
This is just my curiosity but do we need to use this if def for test cases? This is because when disabling FLB_HAVE_METRICS, the defined test cases are not effective under this condition.

Reply:
hmm maybe we don't need it... I don't think metrics are optional anymore